java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/JavaParserFacade.java

```diff
@@ -531,6 +527,14 @@ private Type getTypeConcrete(Node node, boolean solveLambdas) {
         }
     }
 
+    public Type convertToUsageVariableType(VariableDeclarationExpr var) {
+        Type type = JavaParserFacade.get(typeSolver).convertToUsage(var.getElementType(), var);
+        for (int i=0;i<var.getArrayBracketPairsAfterElementType().size();i++) {
+            type = new ArrayType(type);
```

Comment:
Currently I am rewriteing javaparser-symbol-model, then I plan to heavily refactor all of this code. All the tests should permit to do some refactoring with confidence.

But I will start fixing this :D